File: minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i=0,j=0,sum=0,n=nums.size();
        int ans=n+1;
        while(j<n){
            while(j<n && sum<target){
                sum+=nums[j++];
            }
            while(i<j && sum>=target){
                ans=min(ans,j-i);
                sum-=nums[i++];
            }
        }
        return ans==n+1 ? 0 : ans;
    }
};

File: minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        i=0
        j=0
        Sum=0
        n=len(nums)
        ans=n+1
        while(j<n):
            while(j<n and Sum<target):
                Sum=Sum+nums[j]
                j=j+1
            while(i<j and Sum>=target):
                ans=min(ans,j-i)
                Sum=Sum-nums[i]
                i=i+1
        if(ans==n+1):
            return 0
        return ans